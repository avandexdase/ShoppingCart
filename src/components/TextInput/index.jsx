import React from 'react';

function TextInput({
  field: { name, value, onChange, onBlur },
  form: { touched, errors },
  label,
  ...props
}) {
  return (
    <div className="form-field">
      <div className="form-field__control">
        <input
          type="text"
          className="form-field__input"
          placeholder=" "
          name={name}
          id={name}
          value={value}
          onChange={onChange}
          onBlur={onBlur}
          {...props}
        />
        <label htmlFor={name} className="form-field__label">
          {label}
        </label>
        {touched[name] && errors[name] && (
          <p className="form-field__error">{errors[name]}</p>
        )}
      </div>
      <div className="form-field__bar"></div>
    </div>
  );
}

export default TextInput;
