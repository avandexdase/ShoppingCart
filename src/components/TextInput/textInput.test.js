import React from 'react';
import renderer from 'react-test-renderer';
import TextInput from './index';

it('<TextInput />', () => {
  const onChangeHandler = jest.fn();
  const onBlurHandler = jest.fn();
  const field = {
    name: 'abc',
    value: 'ram',
    onChange: onChangeHandler,
    onBlur: onBlurHandler,
  };
  const form = { touched: 'false', errors: '' };
  const label = 'name';
  const component = renderer.create(
    <TextInput field={field} form={form} label={label} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
