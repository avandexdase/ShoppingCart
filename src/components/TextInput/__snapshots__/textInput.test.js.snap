// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> 1`] = `
<div
  className="form-field"
>
  <div
    className="form-field__control"
  >
    <input
      className="form-field__input"
      id="abc"
      name="abc"
      onBlur={[MockFunction]}
      onChange={[MockFunction]}
      placeholder=" "
      type="text"
      value="ram"
    />
    <label
      className="form-field__label"
      htmlFor="abc"
    >
      name
    </label>
  </div>
  <div
    className="form-field__bar"
  />
</div>
`;
